usdQt/hierarchyModel.py

```diff
@@ -0,0 +1,457 @@
+        if not columns:
+            # By default show all possible columns.
+            self.columns = [HierarchyStandardModel.Name,
+                            HierarchyStandardModel.Type,
+                            HierarchyStandardModel.Kind]
+        else:
+            self.columns = columns
+
+    def headerData(self, section, orientation, role):
+        if role == QtCore.Qt.DisplayRole:
+            return self.columns[section]
+
+        return super(HierarchyStandardModel, self).headerData(section, orientation, role)
+
+    def columnCount(self, parent):
+        return len(self.columns)
+
+    def data(self, modelIndex, role=QtCore.Qt.DisplayRole):
+        if not (modelIndex.isValid()):
+            return None
+        column = self.columns[modelIndex.column()]
+        if role == QtCore.Qt.ForegroundRole:
+            prim = self._GetPrimForIndex(modelIndex)
+            brush = HierarchyStandardModel.NormalColor
+            if not prim.IsActive():
+                brush = QtGui.QBrush(brush)
+                brush.setColor(brush.color().darker(
+                    HierarchyStandardModel.InactiveDarker))
+            return brush
+        elif role == QtCore.Qt.DecorationRole:
+            if modelIndex.column() == 0:
+                prim = self._GetPrimForIndex(modelIndex)
+                if prim.HasAuthoredInherits() or \
+                        prim.HasAuthoredReferences() or \
+                        prim.HasVariantSets() or \
+                        prim.HasPayload() or \
+                        prim.HasAuthoredSpecializes():
+                    return utils.IconCache.Get(self.ArcsIconPath)
+                else:
+                    return utils.IconCache.Get(self.NoarcsIconPath)
+        elif role == QtCore.Qt.DisplayRole:
+            if column == HierarchyStandardModel.Name:
+                return super(HierarchyStandardModel, self).data(modelIndex, role)
+            elif column == HierarchyStandardModel.Type:
+                prim = self._GetPrimForIndex(modelIndex)
+                typeName = prim.GetTypeName()
+                return typeName if typeName else ""
+            elif column == HierarchyStandardModel.Kind:
+                prim = self._GetPrimForIndex(modelIndex)
+                kind = prim.GetMetadata('kind')
+                return kind if kind else ""
+            else:
+                raise Exception("shouldn't happen")
+        elif role == QtCore.Qt.ToolTipRole:
+            prim = self._GetPrimForIndex(modelIndex)
+            specifier = prim.GetSpecifier()
+            primType = prim.GetTypeName()
+            documentation = prim.GetDocumentation()
+            if specifier == Sdf.SpecifierDef:
+                toolTipString = "Defined %s"
+            elif specifier == Sdf.SpecifierOver:
+                toolTipString = "Undefined %s"
+            elif specifier == Sdf.SpecifierClass:
+                toolTipString = "Abstract %s"
```

Comment:
I don't know offhand if these are the only specifiers allowed, but even if that's the case for the sake of clarity, and to a lesser degree forward-compatibility, we should probably have an `else` here to either raise an error (`Exception("this should never happen!)`) or provide a fallback `toolTipString = "Unknown %s"`.